reader.go

```diff
@@ -708,12 +710,56 @@ func NewReader(config ReaderConfig) *Reader {
 		version: version,
 	}
 	if r.useConsumerGroup() {
+		kafkaDialer := config.Dialer
+		dialer := (&net.Dialer{
+			Timeout:       kafkaDialer.Timeout,
+			Deadline:      kafkaDialer.Deadline,
+			LocalAddr:     kafkaDialer.LocalAddr,
+			DualStack:     kafkaDialer.DualStack,
+			FallbackDelay: kafkaDialer.FallbackDelay,
+			KeepAlive:     kafkaDialer.KeepAlive,
+		})
+
+		var resolver Resolver
+		if r, ok := kafkaDialer.Resolver.(*net.Resolver); ok {
+			dialer.Resolver = r
+		} else {
+			resolver = kafkaDialer.Resolver
+		}
+
+		// For backward compatibility with the pre-0.5 APIs, support custom
+		// resolvers by wrapping the dial function.
+		dial := func(ctx context.Context, network, addr string) (net.Conn, error) {
+			start := time.Now()
+			defer func() {
+				r.stats.dials.observe(1)
+				r.stats.dialTime.observe(int64(time.Since(start)))
+			}()
+			address, err := lookupHost(ctx, addr, resolver)
+			if err != nil {
+				return nil, err
+			}
+			return dialer.DialContext(ctx, network, address)
+		}
+
+		transport := &Transport{
+			Dial:     dial,
+			SASL:     kafkaDialer.SASLMechanism,
+			TLS:      kafkaDialer.TLS,
+			ClientID: kafkaDialer.ClientID,
+		}
+
+		if transport.ClientID == "" {
+			transport.ClientID = DefaultClientID
+		}
+
+		r.transport = transport
```

Comment:
We do something similar in `kafka.Writer` I believe, would it be worth abstracting this logic internally?